yt_dlp/extractor/tiktok.py

```diff
@@ -511,23 +535,30 @@ def _real_extract(self, url):
         try:
             return self._extract_aweme_app(video_id)
         except ExtractorError as e:
-            self.report_warning(f'{e}; Retrying with webpage')
+            self.report_warning(f'{e.orig_msg}; Retrying with webpage')
 
-        # If we only call once, we get a 403 when downlaoding the video.
-        self._download_webpage(url, video_id)
-        webpage = self._download_webpage(url, video_id, note='Downloading video webpage')
+        # If UA starts with `Mozilla/5.0`, but shorter than 26 character, 403 response is received
+        webpage = self._download_webpage(
+            url, video_id, headers={'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:98.0) Gecko/20100101 Firefox/98.0'})
         next_data = self._search_nextjs_data(webpage, video_id, default='{}')
 
         if next_data:
             status = traverse_obj(next_data, ('props', 'pageProps', 'statusCode'), expected_type=int) or 0
             video_data = traverse_obj(next_data, ('props', 'pageProps', 'itemInfo', 'itemStruct'), expected_type=dict)
         else:
-            sigi_json = self._search_regex(
-                r'>\s*window\[[\'"]SIGI_STATE[\'"]\]\s*=\s*(?P<sigi_state>{.+});',
-                webpage, 'sigi data', group='sigi_state')
-            sigi_data = self._parse_json(sigi_json, video_id)
-            status = traverse_obj(sigi_data, ('VideoPage', 'statusCode'), expected_type=int) or 0
-            video_data = traverse_obj(sigi_data, ('ItemModule', video_id), expected_type=dict)
+            try:
+                sigi_data = get_element_by_id('SIGI_STATE', webpage)
+            except ValueError:
+                sigi_data = self._search_regex(
+                    '(?s)<script\\s[^>]*?\\b'
+                    'id\\s*=\\s*(?P<q>"|\'|\\b)SIGI_STATE(?P=q)'
+                    '[^>]*>\\s*'
+                    '(?P<json>\\{.+?\\})\\s*(?:;[^<]+)?'
+                    '</script', webpage, 'sigi data', group='json')
```

Comment:
```suggestion
                sigi_data = self._search_regex(
                    r'(?s)<script\s[^>]*\bid\s*=\s*(?P<q>["\'])SIGI_STATE(?P=q)[^>]*>\s*(?P<json>{.+?})\s*;',
                    webpage, 'sigi data', group='json')
```
Give an example URL where this regex is needed

Reply:
There's none among test cases. It's only possible if `webpage` contains invalid tag afak.
Should it be safeguarded by adding (_unexpected_) warning to `except` clause?

Reply:
> It's only possible if webpage contains invalid tag afak.

and this is known to happen? If so, provide an example URL where this code is reached. if not, I do not understand why this is needed. This is looking for the same thing as the above get_element_by_attribute. Are you suggesting that all code that uses `get_element_by_attribute` must have a fallback? That makes no sense.